bendigobank.rb

```diff
@@ -0,0 +1,90 @@
+require_relative 'account.rb'
+require_relative 'transaction.rb'
+
+class Bendigobank
+  URL = "http://demo.bendigobank.com.au"
+
+  def self.execute
+    connect
+    fetch_accounts
+    fetch_transactions
+  end
+
+  def self.connect
+    Watir.logger.ignore :deprecations
+    @browser = Watir::Browser.new :firefox
+    @browser.goto URL
+    @browser.button(name: 'customer_type').click
+  end
+
+  def self.fetch_accounts
+    html = Nokogiri::HTML.fragment(@browser.ol(class: "grouped-list grouped-list--compact").html)
+    parse_accounts(html)
+  end
+
+  def self.fetch_transactions
+    to_date = Time.new().to_datetime
+    from_date = to_date << 2
+    if from_date.month < 10
+      from_date = from_date.day.to_s + "/" + "0" + from_date.month.to_s + "/" + from_date.year.to_s
+    end
+    if to_date.month < 10 && to_date.day < 10
+      to_date = "0" + to_date.day.to_s + "/" + "0" + to_date.month.to_s + "/" + to_date.year.to_s
+      from_date = "0" + from_date.day.to_s + "/" + from_date.month.to_s + "/" + from_date.year.to_s
+    elsif to_date.month < 10
+      to_date = to_date.day.to_s + "/" + "0" + to_date.month.to_s + "/" + to_date.year.to_s
+    elsif date.day < 10
+      to_date = "0" + to_date.day.to_s + "/" + to_date.month.to_s + "/" + to_date.year.to_s
+      from_date = "0" + from_date.day.to_s + "/" + from_date.month.to_s + "/" + from_date.year.to_s
+    end
+     @browser.ol(class: 'grouped-list__group__items').each do |li|
+       li.a(class: 'panel--hover').click
+       @browser.i(class: 'ico-nav-bar-filter_16px').click
+       @browser.a(class: 'panel--bordered__item').click
+       @browser.ul(class: 'radio-group').li(index: 8).click
+       @browser.text_field(name: 'toDate').set(to_date)
+       @browser.text_field(name: 'fromDate').set(from_date)
+       @browser.button(class: 'button--primary').click
+       @browser.button(class: 'button--primary').click
+       while @browser.div(class: '_3Wd5wOSiEN').present? != true
+         if @browser.div(class: 'full-page-message').present? == true
+           break
+         end
+         @browser.scroll.to :bottom
+       end
+       html = Nokogiri::HTML.fragment(@browser.div(class: 'activity-container').html)
+       account_name = @browser.h2(class: 'yBcmat9coi').text
+       parse_transactions(html,account_name)
+     end
+  end
+
+  def self.parse_accounts(html)
+     html.css('.grouped-list__group__items li').each do |li|
+      name = li.css('._3jAwGcZ7sr').text
+      balance = li.css('.S3CFfX95_8').text
+      currency = balance[19]
+      nature = "credit_card"
+      transactions = Array[]
+      @account = Accounts.new(name,currency,balance[20..-1],nature,transactions).to_hash
+     end
+  end
+
+  def self.parse_transactions(html,account_name)
+    html.css('.grouped-list--indent').css('.grouped-list__group').each do |li|
+      date = li.css('.grouped-list__group__heading').text
+      li.css('.grouped-list__group__items').css('._2EcJACN7jc').each do |li|
```

Comment:
Find a universal solution for your code do not use hardcoded values `'._2EcJACN7jc'`